src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -72,8 +73,19 @@ public OperationTypes getLongRunningOperation(Method method) {
   /* Return a list of method signatures, aka flattenings, specified on a given method.
    * This flattens the repeated additionalSignatures into the returned list of MethodSignatures. */
   public List<MethodSignature> getMethodSignatures(Method method) {
+    Optional<FieldDescriptor> signatureDescriptor =
+        method
+            .getOptionFields()
+            .keySet()
+            .stream()
+            .filter(o -> o.getFullName().equals("google.api.method_signature"))
+            .findAny();
+    if (!signatureDescriptor.isPresent()) {
+      return ImmutableList.of();
+    }
     MethodSignature methodSignature =
-        method.getDescriptor().getMethodAnnotation(AnnotationsProto.methodSignature);
```

Comment:
Does this NPE, or return null? If it returns null, it seems simpler than using streams, and doesn't hard-code the "google.api.method_signature" string.

Reply:
The problem with the previous implementation is that `MethodSignature methodSignature` would always return a non-null object, which makes it impossible to differentiate between an empty method signature (with 0 parameters) and the non-presence of a method signature. 

We need to do the stream thing to see if the method_signature annotation is even present.

Added a code comment to explain.

Reply:
Hmm, it is a bit annoying that we can't access the underlying MethodDescriptorProto, as then we could call getOptions().hasExtension(...), similar to how the implementation of getMethodAnnotation calls getOptions().getExtension(...).

Does getOptionsFields().contains(AnnotationsProto.methodSignature.defaultInstance()) work?

Reply:
Oh that does work! Thanks! It's a lot cleaner.
Also added a test for empty method signatures.